src/amo/components/AddonTitle/index.js

```diff
@@ -16,53 +20,73 @@ type Props = {|
 type InternalProps = {|
   ...Props,
   i18n: I18nType,
+  isRTL: boolean,
 |};
 
-export const AddonTitleBase = ({ addon, i18n }: InternalProps) => {
-  let children;
-  let htmlTitle;
+export const AddonTitleBase = ({ addon, i18n, isRTL }: InternalProps) => {
+  const authors = [];
 
-  if (addon) {
-    let title;
-    if (!addon.authors) {
-      title = addon.name;
-    } else {
-      const authorList = addon.authors.map((author) => {
-        if (author.url) {
-          return `<a href="${author.url}">${author.name}</a>`;
-        }
+  if (addon && addon.authors) {
+    const addonAuthors = addon.authors;
 
-        return author.name;
-      });
+    // translators: A comma, used in a list of authors: a1, a2, a3.
+    const comma = i18n.gettext(',');
+    const separator = isRTL ? ` ${comma}` : `${comma} `;
 
-      title = i18n.sprintf(
-        // translators: Example: The Add-On <span>by The Author</span>
-        i18n.gettext('%(addonName)s %(startSpan)sby %(authorList)s%(endSpan)s'),
-        {
-          addonName: addon.name,
-          authorList: authorList.join(', '),
-          startSpan: '<span class="AddonTitle-author">',
-          endSpan: '</span>',
-        },
+    addonAuthors.forEach((author, index) => {
+      authors.push(
+        author.url ? (
+          <Link key={author.id} to={`/user/${author.username}/`}>
+            {author.name}
+          </Link>
+        ) : (
+          author.name
+        ),
       );
-    }
 
-    htmlTitle = sanitizeHTML(title, ['a', 'span']);
-  } else {
-    children = <LoadingText width={70} />;
+      if (index + 1 < addonAuthors.length) {
+        authors.push(separator);
+      }
+    });
   }
 
   return (
-    <h1
-      className="AddonTitle"
-      // eslint-disable-next-line react/no-danger
-      dangerouslySetInnerHTML={htmlTitle}
-    >
-      {children}
+    <h1 className="AddonTitle">
+      {addon ? (
+        <React.Fragment>
+          {addon.name}
+          {authors.length > 0 && (
+            <span className="AddonTitle-author">
+              {isRTL ? (
+                <React.Fragment>
+                  {authors}{' '}
+                  {// translators: Example: add-on "by" some authors
+                  i18n.gettext('by')}
+                </React.Fragment>
+              ) : (
+                <React.Fragment>
+                  {i18n.gettext('by')} {authors}
```

Comment:
This should probably have a translators hint as well.

Having said that, are you sure this will work for all languages? You've accounted for RTL, but is it possible that some language wouldn't use a space as a separator? You're still dictating what will appear between the "by" and the list of authors and I'm a bit concerned that that still might be an issue. I feel like we need to have them translate the full "by " (i.e., "by" plus a space), but I could be wrong about that. I realize it probably will make the code trickier to allow for that.

Reply:
Those spaces are "UI" spaces and not "writing" spaces if that makes sense. We need an empty space between text and components/HTML. I could have used `&nbsp;` but decided to only use a space.

Reply:
> I feel like we need to have them translate the full "by " (i.e., "by" plus a space), but I could be wrong about that

That was my initial idea but `gettext()` does not let you do that anyway 😞 It is still an improvement over what we have right now though

Reply:
> This should probably have a translators hint as well.

I don't know because the other one has a hint already. So I thought it would create only one entry, and not two, but sure I can add a comment.